cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -449,6 +454,73 @@ angular.module(PKG.name + '.commons')
       DAGPlusPlusNodesActionsFactory.setConnections(connections);
     }
 
+    function resetEndpointsAndConnections() {
+      // have to unbind and bind again, otherwise calling detachEveryConnection will call formatConnections
+      // for every connection that we detach
+      vm.instance.unbind('connection');
+      vm.instance.unbind('connectionDetached');
+      endpoints = [];
+      vm.instance.detachEveryConnection();
+      vm.instance.deleteEveryEndpoint();
+      addEndpoints();
+      addConnections();
+      vm.instance.bind('connection', formatConnections);
+      vm.instance.bind('connectionDetached', formatConnections);
+    }
```

Comment:
I tried doing that, but it seems like because I'm getting the nodes/connections from the store when I do Reset (instead of organically changing them through jsPlumb), jsPlumb doesn't know how to render them. That's why I have to call these functions manually.